caom2pipe/caom_composable.py

```diff
@@ -1136,6 +1135,32 @@ def update(self, file_info):
         self._logger.debug('End update')
         return self._observation
 
+    def _init_read_groups(self, config):
+        for entry in config.data_read_groups:
+            self._data_read_groups.add(entry)
+        for entry in config.meta_read_groups:
+            self._meta_read_groups.add(entry)
+
+    def _update_artifact(self, artifact):
+        """
+        :param artifact: Artifact instance
+        :return:
+        """
+        return
+
+    def _update_plane(self, plane):
+        self._update_groups(plane.data_read_groups, self._data_read_groups)
+        self._update_groups(plane.meta_read_groups, self._meta_read_groups)
+
+    @staticmethod
+    def _update_groups(replace_these, with_these):
+        if len(with_these) > 0:
```

Comment:
If this method supposed to clone `with_these` it can be done simply with `replace_these = with_these.copy()`.
And it shouldn't pre-check the size unless there's an explicit mechanism to clean a groups list.

Reply:
I could be doing it wrong, but using `replace_these = with_these.copy()` results in this error: `AttributeError: 'URISet' object has no attribute 'copy'`?

Reply:
Oops! Forgot this is a `TypedSet` that doesn't come with this functionality or with a `clear` method. The only thing you could do is to replace the second while loop with `replace_these.update(with_these)` but it doesn't buy you much and I don't think it's worth it so please ignore my comment.